app/decorators/confirmation_decorator.rb

```diff
@@ -110,6 +113,20 @@ def excise_additional_codes
     user_session.excise_additional_code.values.join(', ')
   end
 
+  def additional_codes
+    (user_session.additional_code_uk.values + user_session.additional_code_xi.values).compact.join(', ')
+  end
+
+  def document_code_for(value)
+    return nil unless value.values.reject(&:empty?).any?
```

Comment:
Really minor code style but this line isn't the most straight forward to read? An `if` would remove the double negative and renaming the method arg could make it clearer?

```ruby
def document_code_for(????)
  return nil if ????.values.reject(&:empty?).empty?

  document_codes
end

Reply:
Valid points! Will update.